app/src/main/java/com/example/myfitnessbuddy/fragments/interior/HomeFragment.java

```diff
@@ -31,31 +60,257 @@ protected int getFragmentLayout() {
+                    }
+
+                }
+
+                ids.clear();
+                acceptedIds.clear();
+                declinedIds.clear();
+                removedIds.clear();
+
+                for(DataSnapshot snapshot : dataSnapshot.child(Constants.MATCHES).getChildren()){
+                    if(userType.equals(Constants.TRAINEE)){
+                        if(snapshot.child(Constants.TRAINEE_ID).getValue().toString().equals(currentUserId)){
+                            String relStatus = snapshot.child(Constants.RELATIONSHIP_STATUS).getValue().toString();
+                            String trainerId = snapshot.child(Constants.TRAINER_ID).getValue().toString();
+                               if(relStatus.equals(Constants.NOT_ACCEPTED_STATUS)){
+                                   ids.add(trainerId);
+                               }
+                               else if(relStatus.equals(Constants.ACCEPTED_STATUS)){
+                                   acceptedIds.add(trainerId);
+                               }
+                               else if (relStatus.equals(Constants.DECLINED)){
+                                   declinedIds.add(trainerId);
+                               }
+                               else if (relStatus.equals(Constants.REMOVED_BY_TRAINER_STATUS)){
+                                   removedIds.add(trainerId);
+                               }
+                        }
+                    }
+                    else if(userType.equals(Constants.TRAINER)){
+                        if(snapshot.child(Constants.TRAINER_ID).getValue().toString().equals(currentUserId)){
+                            String relStatus = snapshot.child(Constants.RELATIONSHIP_STATUS).getValue().toString();
+                            String traineeId = snapshot.child(Constants.TRAINEE_ID).getValue().toString();
+
+                            if(relStatus.equals(Constants.NOT_ACCEPTED_STATUS)
+                                    || relStatus.equals(Constants.TRAINEE_ACCEPTED_STATUS)){
+                                Log.i("Noemi","Traine ids notif: " + snapshot.child(Constants.TRAINEE_ID).getValue().toString());
+                                ids.add(snapshot.child(Constants.TRAINEE_ID).getValue().toString());
+                            }
+                            else if (relStatus.equals(Constants.REMOVED_BY_TRAINEE_STATUS)){
+                                removedIds.add(traineeId);
+                            }
+
+                        }
+                    }
+
+                }
+
+                if(!acceptedIds.isEmpty()){
+                    for(String s : acceptedIds){
+                        traineeNotifications.add(new NotificationData(Constants.ACCEPTANCE_TITLE,Constants.ACCEPTANCE_MESSAGE + s,Constants.NOTIFYING_TYPE,s));
+                        notifications = traineeNotifications;
```

Comment:
daca intotdeauna in aplicatie va fi doar Trainer sau doar Trainee, cred ca e destul sa ai notifications.add(...), din punctul asta pana unde folosesti ultima data notifications. Adica fara traineeNotifications sau trainerNotifications. 

Doar sa ai grija sa fie initializat notifications inainte sa adaugi la el